src/lambda-edge/refresh-auth/index.ts

```diff
@@ -20,14 +20,21 @@ export const handler: CloudFrontRequestHandler = async (event) => {
 
         validateRefreshRequest(currentNonce, originalNonce, idToken, accessToken, refreshToken);
 
+        let headers: { 'Content-Type': string, Authorization?: string } = { 'Content-Type': 'application/x-www-form-urlencoded' }
+
+        if(clientSecret !== '') {
+            const encodedSecret = Buffer.from(`${clientId}:${clientSecret}`).toString('base64');
+            headers['Authorization'] = `Basic ${encodedSecret}`
+        }
+
         let tokens = { id_token: idToken!, access_token: accessToken!, refresh_token: refreshToken! };
         try {
             const body = stringifyQueryString({
                 grant_type: 'refresh_token',
                 client_id: clientId,
                 refresh_token: refreshToken,
             });
-            const res = await httpPostWithRetry(`https://${cognitoAuthDomain}/oauth2/token`, body, { headers: { 'Content-Type': 'application/x-www-form-urlencoded' } });
+            const res = await httpPostWithRetry(`https://${cognitoAuthDomain}/oauth2/token`, body, { headers: headers });
```

Comment:
Just do:

```
{ headers }
```

Do it that way everywhere in the code

Reply:
Updated